test/xchainGauges/rootGaugeV2.ts

```diff
@@ -0,0 +1,302 @@
+      const gaugeControllerAddress = (
+        await getWithName("GaugeController", "mainnet")
+      ).address
+      gaugeController = await ethers.getContractAt(
+        "GaugeController",
+        gaugeControllerAddress,
+      )
+
+      const minterAddress = (await getWithName("Minter", "mainnet")).address
+      sdl = await ethers.getContractAt("SDL", sdlAddress)
+      vesdl = await ethers.getContractAt("VotingEscrow", veSDLAddress)
+      minter = await ethers.getContractAt("Minter", minterAddress)
+
+      const contracts = await setupAnyCallTranslator(users[0], ANYCALL_ADDRESS)
+      anyCallTranslator = contracts.anyCallTranslator
+
+      // **** Setup rootGauge Factory ****
+      rootGaugeFactory = await setupRootGaugeFactoryV2(
+        anyCallTranslator.address,
+        users[0],
+        false,
+        sdlAddress,
+        gaugeControllerAddress,
+        minterAddress,
+      )
+      await anyCallTranslator.addKnownCallers([rootGaugeFactory.address])
+
+      // Deploy and set ArbitrumBridger for forked mainnet testing
+      const bridgerFactory = await ethers.getContractFactory("ArbitrumBridger")
+      arbitrumBridger = (await bridgerFactory.deploy(
+        GAS_LIMIT,
+        GAS_PRICE,
+        sdl.address,
+      )) as ArbitrumBridger
+      await rootGaugeFactory.set_bridger(
+        CHAIN_ID.ARBITRUM_MAINNET,
+        arbitrumBridger.address,
+      )
+
+      await rootGaugeFactory.deploy_gauge(
+        CHAIN_ID.ARBITRUM_MAINNET,
+        convertGaugeNameToSalt(TEST_GAUGE_NAME),
+        TEST_GAUGE_NAME,
+      )
+
+      rootGauge = await ethers.getContractAt(
+        "RootGaugeV2",
+        await rootGaugeFactory.get_gauge(CHAIN_ID.ARBITRUM_MAINNET, 0),
+      )
+    },
+  )
+
+  beforeEach(async () => {
+    await setupTest()
+  })
+
+  after(async () => {
+    await network.provider.request({
+      method: "hardhat_reset",
+      params: [],
+    })
+  })
+
+  describe("initialize", () => {
+    it(`Can't initialize twice`, async () => {
+      await expect(
+        rootGauge.initialize(
+          arbitrumBridger.address,
+          CHAIN_ID.ARBITRUM_MAINNET,
+          TEST_GAUGE_NAME,
+        ),
+      ).to.be.revertedWith("already initialized")
+    })
+
+    it(`Successfully sets storage variables`, async () => {
+      expect(await rootGauge.bridger()).to.equal(arbitrumBridger.address)
+      expect(await rootGauge.chain_id()).to.equal(CHAIN_ID.ARBITRUM_MAINNET)
+      expect(await rootGauge.name()).to.equal(
+        "Saddle " + TEST_GAUGE_NAME + " Root Gauge",
+      )
+      expect(await rootGauge.factory()).to.equal(rootGaugeFactory.address)
+      const inflationParams = await rootGauge.inflation_params()
+      expect(inflationParams.rate).to.equal("2066798941798941798")
+      expect(inflationParams.finish_time).to.equal("1664226306")
+      expect(await rootGauge.last_period()).to.equal("2750")
+    })
+
+    it(`Successfully sets storage variables after kill`, async () => {
```

Comment:
Test names could be renamed to reflect the tests better
`Calling user_checkpoint after killing the gauge sets inflation_params.rate to 0`

Reply:
Also this test seems like it is under `initialize function` which doesnt seem to fit for testing for `set_killed function`